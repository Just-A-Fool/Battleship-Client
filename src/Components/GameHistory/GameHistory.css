.game-history-list {
    border: 0.1rem solid white; 
    width: 90%;
    margin: 0 auto;
    padding: 10px;
    border-radius: 1rem;
    background-color: rgb(30, 52, 90);
}

.gameHistory h1 {
    margin-bottom: 2rem;
}

.gameHistory {
    margin-top: 8rem;
}

.game-history-li {
    display: flex;
    border-bottom: 1px solid white;
    width: 80%;
    margin: 0 auto;
}

.game-history-region {
    width: 50%;
    /* text-align: left; */
    /* border:1px solid red; */
}

.player-Win {
    color: greenyellow;
    font-weight: bold;
}

.player-Loss {
    color: #FF0100;
    font-weight: bold;
}

.historyWelcome {
    font-size: 1.5rem;
    text-align: left;
    margin-left: 1rem;
    margin-top: 8.3rem;
    margin-bottom: 2rem;
}

@media screen and (min-width: 900px) {
    .game-history-list {
        width: 75%;
    }
}

@media screen and (min-width: 1025px) {
    .game-history-list {
        width: 65%;
    }

    .gameHistory {
        margin-top: 9rem;
    }

    .historyWelcome {
        margin-bottom: 3rem;
    }
}
