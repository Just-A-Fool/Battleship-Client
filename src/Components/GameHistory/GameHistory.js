import React from 'react';
import { withRouter } from 'react-router-dom';
import LoadGameApiService from '../../Services/load-game-api-service';
import './GameHistory.css';
import Button from '../Button/Button';
import Header from '../Header/Header';
import './GameHistory.css';
import Footer from '../Footer/Footer';

class GameHistory extends React.Component {
    state = {
        error: null,
    }

    setError = (err) => {
        this.setState({
            error: err.error,
            history: null
        })
    }

    componentDidMount = () => {
        LoadGameApiService.getGameHistory()
            .then(data => {
                this.setState({
                    history: data
                })
            })
    }

    handleInspect = (player, gameId, playerUsername, opponentUsername, playerWin) => {
        this.props.setResults(player, gameId, playerUsername, opponentUsername, playerWin);
        this.props.history.push('/result');
    }

    render() {
        let completedGames = 'No game history availiable';
        let userWelcome = null
        if (this.state.history) {
            userWelcome = <h2 className='dashboardWelcome'><span className='username'>{this.state.history.playerUsername}</span>'s Game History</h2>

            completedGames = this.state.history.result.map(game => {
                let playerUsername = this.state.history.playerUsername;
                let opponentUsername = game.player1_username === playerUsername ? game.player2_username : game.player1_username;


                let winnerUsername = game[`${[game.winner]}_username`];
                let winBool = playerUsername === winnerUsername;
                let winStatus = winBool ? 'Win' : 'Loss';


                let playerString = playerUsername === game.player1_username ? 'player1' : 'player2'

                let winReason = null;

                if (game.game_status === 'complete') winReason = 'skill';
                else if (game.game_status === 'expired') winReason = 'default';
                else if (game.game_status === 'forfeited') winReason = 'forfeit';

                return (
                    <li key={game.game_id} className='game-history-li'>
                        <div className='game-history-region'>
                            <span className={`player-${winStatus}`}>{winStatus} vs {opponentUsername}</span>
                            <p>{winnerUsername} won by {winReason}</p>
                        </div>
                        <div className='game-history-region'>
                            <Button onClick={() => this.handleInspect(playerString, game.game_id, playerUsername, opponentUsername, winBool)}>Inspect</Button>
                        </div>
                    </li>
                );
            })
        }

        return (
            <div className='gameHistory'>
                <Header />
                {userWelcome}
                <h2>Game History</h2>
                
                <ul className='game-history-list'>
                    {completedGames}
                </ul>

                <Footer />
            </div>
        );
    }
}

export default withRouter(GameHistory);


