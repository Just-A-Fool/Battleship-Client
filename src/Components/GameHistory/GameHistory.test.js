import React from 'react';
import ReactDOM from 'react-dom';
import GameHistory from './GameHistory';
import { mount } from 'enzyme';
import toJson from 'enzyme-to-json';
import { BrowserRouter as Router } from 'react-router-dom';


describe('GameHistory component', () => {
  it('Renders without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(<Router><GameHistory /></Router>, div);
    ReactDOM.unmountComponentAtNode(div);
  });

  it(`Displays the 'GameHistory' component when rendered`, () => {
    const wrapper = mount(<Router><GameHistory /></Router>)
    expect(toJson(wrapper)).toMatchSnapshot()
  });


  it(`Displays message indicating if no game history is available`, () => {
    const wrapper = mount(<Router><GameHistory /></Router>)
    expect(wrapper.text()).toMatch(/No game history availiable/)
  });

  it(`Contains an 'h1' element for the game history`, () => {
    const wrapper = mount(<Router><GameHistory /></Router>)
    expect(wrapper.contains(<h1>Game History</h1>)).toEqual(true)
  });
  
})