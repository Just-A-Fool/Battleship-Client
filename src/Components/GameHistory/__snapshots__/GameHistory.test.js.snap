// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GameHistory component Displays the 'GameHistory' component when rendered 1`] = `
<BrowserRouter>
  <Router
    history={
      Object {
        "action": "POP",
        "block": [Function],
        "createHref": [Function],
        "go": [Function],
        "goBack": [Function],
        "goForward": [Function],
        "length": 1,
        "listen": [Function],
        "location": Object {
          "hash": "",
          "pathname": "/",
          "search": "",
          "state": undefined,
        },
        "push": [Function],
        "replace": [Function],
      }
    }
  >
    <withRouter(GameHistory)>
      <Route>
        <GameHistory
          history={
            Object {
              "action": "POP",
              "block": [Function],
              "createHref": [Function],
              "go": [Function],
              "goBack": [Function],
              "goForward": [Function],
              "length": 1,
              "listen": [Function],
              "location": Object {
                "hash": "",
                "pathname": "/",
                "search": "",
                "state": undefined,
              },
              "push": [Function],
              "replace": [Function],
            }
          }
          location={
            Object {
              "hash": "",
              "pathname": "/",
              "search": "",
              "state": undefined,
            }
          }
          match={
            Object {
              "isExact": true,
              "params": Object {},
              "path": "/",
              "url": "/",
            }
          }
        >
          <div
            className="gameHistory"
          >
            <withRouter(Header)>
              <Route>
                <Header
                  history={
                    Object {
                      "action": "POP",
                      "block": [Function],
                      "createHref": [Function],
                      "go": [Function],
                      "goBack": [Function],
                      "goForward": [Function],
                      "length": 1,
                      "listen": [Function],
                      "location": Object {
                        "hash": "",
                        "pathname": "/",
                        "search": "",
                        "state": undefined,
                      },
                      "push": [Function],
                      "replace": [Function],
                    }
                  }
                  location={
                    Object {
                      "hash": "",
                      "pathname": "/",
                      "search": "",
                      "state": undefined,
                    }
                  }
                  match={
                    Object {
                      "isExact": true,
                      "params": Object {},
                      "path": "/",
                      "url": "/",
                    }
                  }
                >
                  <div>
                    <Banner>
                      <div
                        className="banner"
                      >
                        <h1>
                          BATTLESHIP
                        </h1>
                      </div>
                    </Banner>
                    <header
                      className="main-header"
                    >
                      <ForwardRef
                        onClick={[Function]}
                      >
                        <button
                          className="Button"
                          onClick={[Function]}
                        >
                           Dashboard
                        </button>
                      </ForwardRef>
                      <ForwardRef
                        onClick={[Function]}
                      >
                        <button
                          className="Button"
                          onClick={[Function]}
                        >
                           Help
                        </button>
                      </ForwardRef>
                      <ForwardRef
                        onClick={[Function]}
                      >
                        <button
                          className="Button"
                          onClick={[Function]}
                        >
                          Logout
                        </button>
                      </ForwardRef>
                    </header>
                  </div>
                </Header>
              </Route>
            </withRouter(Header)>
            <h1>
              Game History
            </h1>
            <ul
              className="game-history-list"
            >
              No game history availiable
            </ul>
            <Footer>
              <footer />
            </Footer>
          </div>
        </GameHistory>
      </Route>
    </withRouter(GameHistory)>
  </Router>
</BrowserRouter>
`;
